tests/src/core/testqgis.cpp

```diff
@@ -543,7 +548,7 @@ void TestQgis::testQgsFlagKeysToValue()
   QCOMPARE( qgsFlagKeysToValue( QString::number( newValue ), defaultValue, true, &ok ), newValue );
   QCOMPARE( ok, true );
   // also try with an invalid int value
-  QCOMPARE( qgsFlagKeysToValue( QString::number( -1 ), defaultValue, true, &ok ), defaultValue );
+  QCOMPARE( qgsFlagKeysToValue( QString::number( std::numeric_limits<int>::quiet_NaN() ), defaultValue, true, &ok ), defaultValue );
```

Comment:
``std::numeric_limits<int>::quiet_NaN()`` evaluates to 0. Is that what you mean ?

Reply:
No, I was sure it evaluates to a NaN value :-1:  
Here the issue is negative number are no more invalid, do you think it will break the test if I use `std::numeric_limits<double>::quiet_NaN()`?

Reply:
I have tried with `std::numeric_limits<long>::max()` and debug it step by step to check if it takes the right code path.

Reply:
I'd avoid using long as it is 32 bit on Windows 64 and 64 bit on other 64-bit systems.  Isn't int enough ?

Reply:
The issue did not come from the string to int conversion but from the call to `metaEnum.valueToKeys( intValue );` with a negative value. Qt6 returns an empty string ("") and Qt5 returns a string representation of a default ("HideReadOnly|OriginProvider|AllTypes")